sklearn/tree/_tree.pyx

```diff
@@ -95,48 +96,129 @@ NODE_DTYPE = np.dtype({
 # =============================================================================
 
 cdef class Criterion:
-    """Interface for impurity criteria."""
+    """Interface for impurity criteria. 
+
+    This object stores methods on how to calculate how good a split is using 
+    different metrics.
+    """
 
     cdef void init(self, DOUBLE_t* y, SIZE_t y_stride, DOUBLE_t* sample_weight,
                    double weighted_n_samples, SIZE_t* samples, SIZE_t start,
                    SIZE_t end) nogil:
-        """Initialize the criterion at node samples[start:end] and
-           children samples[start:start] and samples[start:end]."""
+        """Placeholder for a method which will initialize the criterion.
+
+        Parameters
+        ----------
+        y: array-like, dtype=DOUBLE_t
+            This is a vector of targets in a memory-efficient buffer format, 
+            used to evaluate the split
+        y_stride: SIZE_t
+            This indicates the stride of the buffer to allow access to elements
+            easily
```

Comment:
This is not very informative as it does not provide any information that is not already present in the variable name. Better document something like:

`y` is a 1D buffer that can store values for `n_outputs` target variables. `y_stride` is used to index the kth output value for the ith samples as follows: ``y[i * y_stride + k]``.